eos-gpt2mbr

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash -e
+# Copyright (C) 2015 Endless Mobile, Inc.
+# Licensed under the GPLv2
+
+get_last_char() {
+	ret=$1
+	while [ "${#ret}" != "1" ]; do
+		ret=${ret#?}
+	done
+	echo $ret
+}
+
+# arguments checking
+if [ $# -lt 1 ]; then
+	echo "Usage: $0 <EOS partition>"
+       	exit 0
+fi
+
+root_part=$1
+root_partno=$(get_last_char ${root_part})
+root_disk=${root_part%?}
+
+swap_partno=$((root_partno + 1))
+swap_part=${root_disk}${swap_partno}
+
+# check if we passed the full device file path
+if [ "${root_part//dev/}" = "${root_part}" ]; then
+	echo "eos-gpt2mbr: Specify the full path for the root device file (i.e. /dev/sda3)"
+	exit 0
+fi
+
+# check if we passed the device file for the partition and not for the disk
+case ${root_partno} in
+	*[!0-9]* )
```

Comment:
Should this pattern be `*[!0-9]` (ie no trailing wildcard)? I don't think this is a huge problem though.